dashboard/app/controllers/levels_controller.rb

```diff
@@ -4,7 +4,7 @@
 class LevelsController < ApplicationController
   include LevelsHelper
   include ActiveSupport::Inflector
-  before_filter :authenticate_user!, except: [:embed_blocks, :embed_level]
+  before_filter :authenticate_user!, except: [:show, :embed_blocks, :embed_level]
   before_filter :can_modify?, except: [:show, :index, :embed_blocks, :embed_level]
   skip_before_filter :verify_params_before_cancan_loads_model, only: [:create, :update_blocks]
   load_and_authorize_resource except: [:create, :update_blocks, :edit_blocks, :embed_blocks, :embed_level]
```

Comment:
I checked with @wjordan -- there are some redundancies in the current auth logic here that he's going to clean up in a future commit.  For now the behavior we want is verified by rake tests: non-admin and logged-out users can access the `show`, `embed_blocks` and `embed_level` actions but not any of the modify actions.